### PR TITLE
feat: auto fetch releases

### PR DESCRIPTION
**Pull request by bzp2010**
Changes:

Automatically pull the release records in each APISIX sub-project to generate a release list.

Screenshots of the change:

image

### CODE REVIEW

**Comment by juzhiyuan** [comment]
Great! and how to have a preview on Netlify?

File: scripts/common.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+const languages = ["en", "zh", "es"];
+
+/**
+ * changelog extractor for each version
+ * @param {string} changelogs CHANGELOG.md contents
+ * @param {Object} targetRelease result from GitHub API
+ * @param {string} versionStartsWith version line markdown symbol
+ * @param {boolean} reserveVersionEnds0 reserve the `.0` on version end (e.g. `.0` of `1.5.0`)
+ */
```
Comment (liuxiran) [inline scripts/common.js:62]: Thumbs up to code comments

**Comment by liuxiran** [comment]
also cc @Baoyuantop to have a review, thanks

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -26,15 +26,20 @@ jobs:
...
         with:
-          node-version: "12"
+          node-version: "14"
 
-      - name: Pull Docs
+      - name: Install Dependencies
         run: |
-          node sync-docs.js && git status
+          cd scripts && yarn install
+          cd ../website && yarn install
```
Comment (Baoyuantop) [inline .github/workflows/deploy.yml:34]:
```suggestion
          cd scripts && yarn
          cd ../website && yarn
```

Comment (bzp2010) [inline .github/workflows/deploy.yml:34]: Hi, @Baoyuantop. It seems that the effect of `yarn install` is the same as that of `yarn`, and `yarn install` is more explicit. 🤔

File: scripts/common.js (hunk 1 above)
Comment (bzp2010) [reply scripts/common.js:62]: Haha, and is it accurate? (If I don't make records, I will soon forget the meaning of the parameters)

**Comment by bzp2010** [comment]
> Great! and how to have a preview on Netlify?

Hi, @juzhiyuan. After reviewing and merging it, modify the CI instruction of netlify to display the document content in the preview.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (Baoyuantop) [reply .github/workflows/deploy.yml:34]: Yes, they are equivalent.

**Review by liuxiran — APPROVED** [review]

**Review by Baoyuantop — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**